Chainsaw weapon addition

Jessetriesagain opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
Added a chainsaw to the game. Could be used as a weapon against fellow space stationers or to deconstruct things really fast and aggressively. It's loud, maybe annoyingly so.

This is a beast of a chainsaw - requires two hands to wield as a weapon/tool, use is restricted otherwise.


**Media**
- [X ] I have added screenshots/videos to this PR showcasing its changes in-game

https://github.com/space-wizards/space-station-14/assets/138042477/cc2f7a62-b8f8-4ea3-ae17-c6d0611afe13

**Changelog**

:cl: Jesse
- add: New chainsaw sprites
- add: New chainsaw sounds. It goes vroom.

spectralSpices commented:
The new wield sprite looks good! I just think the blade could use some work-it looks oddly shiny and weirdly shaped.

Aisu9 commented:
no fuel requirement for this new rending monster?

Emisse commented:
make it use fuel

Myakot commented:
Should it even have blunt damage, not purely slashing?

Jessetriesagain commented:
> Should it even have blunt damage, not purely slashing?

No idea! I figured chainsaws sort of "rip and tear" not really slice.

> make it use fuel

I'll take a look at that tomorrow!

> The new wield sprite looks good! I just think the blade could use some work-it looks oddly shiny and weirdly shaped.

Agreed with the shiny bit - I was thinking the same thing and going back and forth. It's a bit too square for my liking when wielding too, I'll see about fixing that shape tomorrow too.

ChilbroBaggins commented:
> Should it even have blunt damage, not purely slashing?

Shredding dmg

UbaserB commented:
the sprite looks a bit… big lol

Emisse commented:
could put this as an expedition reward and let it be op

deltanedas commented:
structural damage would be pro since its a chainsaw
also idling sound so you can hear wheh jason is near you in maints

Jessetriesagain commented:
> structural damage would be pro since its a chainsaw also idling sound so you can hear when jason is near you in maints

I can add the structural damage bit - it does already have idling sounds, but the range is kinda small because I didn't want it to be super annoying.

> the sprite looks a bit… big lol

Yeah I did that on purpose - it's sorta supposed to be a bit of an oversized beast. I'll add that in the description - I actually forgot to do that!

Aisu9 commented:
> could put this as an expedition reward and let it be op

wait... isn't the whole deal with salvage is to kill the powergaming aspect of the job?

asperger-sind commented:
> wait... isn't the whole deal with salvage is to kill the powergaming aspect of the job?

Just make it rare enough so that people can't routinely powergame it

Jessetriesagain commented:
Okay! Made a bunch of adjustments using your feedback!

- Tweak: Made chainsaw blade darker and rounded

- Tweak: Made chainsaw use fuel to do the real damage (sorta wonky because it needs to be "lit" which happens automatically during wield - uses modified Welder template - Sorry, I'm not a programmer 😞 )

- Tweak: Added description to match the size of this monstrosity

- Tweak: Added some extra minor structural damage

- Tweak: Adjusted overall damage (way less blunt) to be not super insane but still insane

https://github.com/space-wizards/space-station-14/assets/138042477/abac4338-8580-42eb-bfaf-16b05cedfab4

Emisse commented:
we will need to genericize the welding bits at some point but i think that's out of scope of this pr

Emisse commented:
you will need to put your sources for the audio in the relevant RGA file, otherwise looks good

Jessetriesagain commented:
> you will need to put your sources for the audio in the relevant RGA file, otherwise looks good

Should be all set - let me know if I misplaced that. I went through documentation as well, but didn't find anything specific.

metalgearsloth reviewed (dismissed):
Needs autoattacks.

Emisse commented on Resources/Audio/Weapons/attributions.yml:
```diff
@@ -8,7 +8,14 @@
 
+
+- files: ["chainsaw.ogg, chainsawidle.ogg"]
+  license: "CC0-1.0"
```
license: "CC0-1.0, https://www.soundjay.com/tos.html"

Emisse commented on Resources/Audio/Weapons/attributions.yml:
```diff
@@ -8,7 +8,14 @@
   source: "https://freesound.org/people/tosha73/sounds/509902/"
 
+
+- files: ["chainsaw.ogg, chainsawidle.ogg"]
```
- files: ["chainsaw.ogg, chainsawidle.ogg, chainsawwield.ogg"]

Jessetriesagain commented:
> Needs autoattacks.

Popped this one to draft instead of a proper pull request. After fumbling around for 4 hours I'm not quite sure how to add auto-attacks here so I think I might just shelve this project for the time being.

Emisse commented:
not gonna block this over autoattacks, you can just fix the things i suggested and its good
